src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -1061,6 +1066,9 @@ private bool TypeExistsOnServer(Type type, bool isGenericArgument)
             if (type.Assembly == typeof(Regex).Assembly) // System.Text.RegularExpressions
                 return true;
 
+            if (type.Assembly == typeof(Constants).Assembly) // Current assembly is Raven.Client
```

Comment:
isnt that dangerous? might cause reindexation in some cases? cc @maciejaszyk @arekpalinski

Reply:
I think this is a bit risky. I'm less concerned about triggering reindexing and more worried about encountering runtime errors

Reply:
donE!